### ISSUE TITLE
Foreign keys names too long (mySQL)

### ISSUE THREAD

**Issue by LazyBoy72** [vapor/fluent#242]
Hi,
I just converted a project I started with Vapor 1.5 into the 2.0.0-beta.22. I'm using mySQL as a database.
All is working good, but I receive this error in the preparation phase:
```
fatal error: Error raised at top level: MySQL Error: Identifier name '_fluent_fk_contact_items.contact_item_type_id-contact_item_types.id' is too long

Identifier: MySQL.MySQLError.1059 (tooLongIdent)
```
What happens is clear, InnoDB storage has a limit of 64 characters for the foreign keys, but there is no nice workaround (at least as far as I got from the source code).
I could rename my models, but I don't want to and I don't think it would be a good design.
Another solution could be to disable the automatic generation of the foreign keys, but this would require me to add them manually everywhere.
I think the best solution is to add a version of the Builder.parent method that accepts a name for the foreign key to be generated. For coherence, this version could throw an error if called when autoForeignKeys is false.

**Comment by LazyBoy72** [comment]
Sorry, I didn't see the pull request.
Closing it myself.

**Comment by tanner0101** [comment]
Thanks for reporting. The solution you mentioned seems good. I will look into this for the 2.0 release.

**Comment by tanner0101** [comment]
Fixed in #251
